lib/Textile/Models/util.ts

```diff
@@ -0,0 +1,80 @@
+import { Timestamp } from 'google-protobuf/google/protobuf/timestamp_pb'
+import { Struct, Value, ListValue } from 'google-protobuf/google/protobuf/struct_pb'
+
+import { pb } from './'
+
+export function convertContactQueryData(data: pb.ContactQuery.AsObject) {
+  const query = new pb.ContactQuery()
+  query.setId(data.id)
+  query.setAddress(data.address)
+  query.setUsername(data.username)
+  return query
+}
+
+export function convertContactQueryOptionsData(data: pb.QueryOptions.AsObject) {
+  const options = new pb.QueryOptions()
+  options.setExcludeList(data.excludeList)
+  options.setFilter(data.filter)
+  options.setLimit(data.limit)
+  options.setLocal(data.local)
+  options.setWait(data.wait)
+  return options
+}
+
```

Comment:
This file includes the conversion functions to go from plain JS object o pb class instance. Not too bad so far. I've only written converted for the types we're already using from pb. This `Directory` one below was quite complex and ended up not being too bad, so let's hope none in the future are much worse.